tlscommon.c

```diff
@@ -626,13 +626,49 @@ static int certattr_matchip(GENERAL_NAME *gn, struct certattrmatch *match){
         && !memcmp(ASN1_STRING_get0_data(gn->d.iPAddress), &match->ipaddr, l)) ? 1 : 0 ;
 }
 
+// Generates a generic DNS. eg: converts idp.openroaming.net to *.openroaming.net
+// Compares generated generic DNS with DNS from certificate
+// returns 1 if both match else returns 0
+static int compareWithModifiedHostname(char* certDNS, char* scriptHostName) {
+    int len = 0;
+    int lenOfGeneric = 0, lenOfModifiedScriptHostName = 0;
+    char* pos;
+    char* modifiedScriptHostName;
+    char *result;
+    char generic[] = "*";
+    pos = strchr(scriptHostName, '.');
+    if(pos == NULL) {
+        return 0;
+    }
+    len = strlen(pos);
+    modifiedScriptHostName = malloc(len + 1);
+    memcpy(modifiedScriptHostName, pos, len);
+    modifiedScriptHostName[len] = '\0';
+    debug(DBG_DBG, "modifiedScriptHostName : %s", modifiedScriptHostName);
+    lenOfGeneric = strlen(generic);
+    lenOfModifiedScriptHostName = strlen(modifiedScriptHostName);
+    result = malloc(lenOfGeneric + lenOfModifiedScriptHostName + 1); 
+    memcpy(result, generic, lenOfGeneric);
+    memcpy(result + lenOfGeneric, modifiedScriptHostName, lenOfModifiedScriptHostName + 1);
```

Comment:
Remove the +1 at the end, since calloc will give you a zero'd array.  If for some reason modifiedScriptHostName isn't null terminated, then you're not corrupting the result buffer along the way.  This guarantees that result will be null terminated.